Doc/library/importlib.rst

```diff
@@ -1584,20 +1584,30 @@ Note that if ``name`` is a submodule (contains a dot),
 Importing a source file directly
 ''''''''''''''''''''''''''''''''
 
+.. note:: ``SourceFileLoader.load_module()`` has been deprecated -- this recipe should be used instead.
```

Comment:
Why the extra note? The method isn't used here and it's documented as deprecated, so why call it out here and add a visible callout in the rendered docs that some might find distracting?

Reply:
I did that because the advice to use SourceFileLoader.load_module() is found all over the internet (e.g. https://krbnite.github.io/How-to-Import-a-Python-Module-from-an-Arbitrary-Path/)
-- and it *seems* a lot simpler than this recipe, and it doesn't raise a deprecation error. so I thought it would be helpful to make it clear why the more complex recipe is preferred. 

That being said, while it might be helpful now, it'll be clutter int he future -- so your call.

Reply:
I think having a lead-in paragraph makes sense, but I think it's pointing at the wrong alternative: the reason importing an arbitrary path is a bad idea is because it's a good way to confuse the import system, and that's why the stdlib's only shorthand API for doing that is deprecated.

The example given uses a top level module/package, so it's potentially OK (although I'm not 100% certain without checking the code for `module_from_spec` - if that bypasses `sys.modules`, you might end up with two different versions of `JSONDecodeError` kicking around, which can easily land you in exception handling hell), but if the file is part of an already importable package than you're far more likely to get some weird state duplication going on (it's a good way to get yourself caught in the ["double import trap"](https://python-notes.curiousefficiency.org/en/latest/python_concepts/import_traps.html#the-double-import-trap)).

The much safer alternative for executing a Python file and getting access to its top-level module state is to use `runpy.run_path` (which returns the global namespace that results from executing the file rather than creating a module object).

In my experience, folks wanting to import from arbitrary paths is usually a classic case of the "XY problem": "X" is "I want to run a Python file from a Python program and get access to the resulting global variables", which becomes the "Y" question of "How do I import a Python module given only its path name?". With `import` being a statement, and the `execfile` builtin ancient history, it's an understandable leap for people to make, but that doesn't mean it's correct when there's no actual *need* to create a module object at all (let alone register it with the import system).

Reply:
> it doesn't raise a deprecation error

... yet. 😉 https://github.com/python/cpython/issues/121604



> I did that because the advice to use `SourceFileLoader.load_module()` is found all over the internet

So is probably outdated Python 2 advice, but that doesn't mean we need to call it out. 😉



> I think having a lead-in paragraph makes sense, but I think it's pointing at the wrong alternative: the reason importing an arbitrary path is a bad idea is because it's a good way to confuse the import system

I agree. I think it would be better to have a sentence or paragraph pointing out that the recipe is an approximation of an `import` statement where you specify the file path. It should also point out that modifying `sys.path` may be better than circumventing the import system or using `runpy.run_path()` if you just need the global namespace and not a module object.

Reply:
FWIW, this is already [in the docs](https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly)! All this PR does is clean up the snippet a little bit.